src/indexes/VideoIndex.js

```diff
@@ -65,232 +56,172 @@ function VideoIndex({
+            <div className="doNotLeaveMessage">
+              There are no videos. Start indexing ones!
             </div>
           </div>
-          <div className="channelPills">
-            <div
-              style={{
-                fontSize: "1.8rem",
-              }}
-            >
-              All Channels in Index{" "}
-            </div>
-            {[...uniqueAuthors].map((author) => (
-              <div key={author + "-" + index} className="channelPill">
-                {author}
-              </div>
-            ))}
+
+          <div className="videoUploadForm">
+            <div className="display-6 m-4">Upload New Videos</div>
+            <UploadYoutubeVideo
+              currIndex={currIndex}
+              taskVideos={taskVideos}
+              setTaskVideos={setTaskVideos}
+              refetchVideos={refetchVideos}
+            />
           </div>
-          <Container fluid className="mb-5">
-            <Row>
-              {videos.data && (
-                <VideoList
-                  index_id={currIndex}
-                  videos={{
-                    data: currentVideos,
-                    isLoading: videos.isLoading,
-                  }}
-                />
-              )}
-              <Container fluid className="my-5 d-flex justify-content-center">
-                <CustomPagination
-                  currentPage={currentPage}
-                  totalPages={totalPages}
-                  onPageChange={setCurrentPage}
-                  nextPage={nextPage}
-                  prevPage={prevPage}
-                />
-              </Container>
-            </Row>
-          </Container>
         </div>
       )}
 
-      {searchPerformed && (
-        <div>
-          {!searchResults.isLoading && searchResults.data.length > 0 && (
-            <div className="searchResultTitle">
-              Search Results for "{searchQuery}"
+      {showVideos && videos && videos.length > 0 && (
+        <ErrorBoundary
+          FallbackComponent={ErrorFallback}
+          onReset={() => refetchVideos()}
+          resetKeys={[keys.VIDEOS]}
+        >
+          <div className="videoUploadForm">
+            <div className="display-6 m-4">Upload New Videos</div>
+            <UploadYoutubeVideo
+              currIndex={currIndex}
+              taskVideos={taskVideos}
+              setTaskVideos={setTaskVideos}
+              refetchVideos={refetchVideos}
+            />
+          </div>
+          {/* Video Search Form */}
+          {!finalSearchQuery && (
+            <div>
+              <div className="videoSearchForm">
+                <div className="title">Search Videos</div>
+                <div className="m-auto p-3 searchFormContainer">
+                  <SearchForm
+                    index={currIndex}
+                    setSearchQuery={setSearchQuery}
+                    searchQuery={searchQuery}
+                    setFinalSearchQuery={setFinalSearchQuery}
+                  />
```

Comment:
Rendering `SearchForm` separately by value of `finalSearchQuery` might cause unexpected rerendering. I suggest you to render this section outside of the conditional section, but if it doesn't look buggy I think we can leave it as it is for now :)

```javascript
            <div>
              <div className="videoSearchForm">
                {!finalSearchQuery && <div className="title">Search Videos</div>}
                <div className="m-auto p-3 searchFormContainer">
                  <SearchForm
                    index={currIndex}
                    setSearchQuery={setSearchQuery}
                    searchQuery={searchQuery}
                    setFinalSearchQuery={setFinalSearchQuery}
                  />
                </div>
              </div>
```